ludwig/data/dataframe/dask.py

```diff
@@ -248,11 +255,68 @@ def to_ray_dataset(self, df):
         return from_dask(df)
 
     def from_ray_dataset(self, dataset) -> dd.DataFrame:
-        return dataset.to_dask()
+        return self._to_dask(dataset)
 
     def reset_index(self, df):
         return reset_index_across_all_partitions(df)
 
+    def _to_dask(
+        self,
+        dataset: Dataset,
+        meta: Union[
+            pd.DataFrame,
+            pd.Series,
+            Dict[str, Any],
+            Iterable[Any],
+            Tuple[Any],
+            None,
+        ] = None,
+    ) -> dd.DataFrame:
+        """Custom Ray to_dask() conversion implementation with meta inference added for compatibility with Ray 2.0
+        and Ray 2.1. Useful for Ray Datasets that have image and audio features.
+
+        TODO(Arnav): Remove in Ray 2.2
+        """
+
+        @dask.delayed
+        def block_to_df(block: Block):
+            if isinstance(block, (ray.ObjectRef, ClientObjectRef)):
+                raise ValueError(
+                    "Dataset.to_dask() must be used with Dask-on-Ray, please "
+                    "set the Dask scheduler to ray_dask_get (located in "
+                    "ray.util.dask)."
+                )
+            block = BlockAccessor.for_block(block)
+            return block.to_pandas()
+
+        # Infer Dask metadata from Datasets schema.
+        schema = dataset.schema()
+        if isinstance(schema, PandasBlockSchema):
+            meta = pd.DataFrame(
+                {
+                    col: pd.Series(dtype=(dtype if not isinstance(dtype, TensorDtype) else np.object_))
+                    for col, dtype in zip(schema.names, schema.types)
+                }
+            )
+        elif pa is not None and isinstance(schema, pa.Schema):
```

Comment:
When would pa be None?  Should we log a warning or error if it is?

Reply:
I'm not sure.

I've gone ahead and removed the check since we're importing `pyarrow as pa` directly, and the import will either succeed for fail.